Release 4.0.0-alpha.1 (#6383)
open-formulieren/open-forms: CHANGELOG.rst

        `latest <https://open-forms.readthedocs.io/en/latest/changelog.html>`_ docs
        version.

4.0.0-alpha.1 (2026-06-22)
==========================

This is an alpha release, meaning it is not finished yet or suitable for production use.

Detailed changes
----------------

> **Breaking changes**

Reviewer: To add:

```rst
* [:backend:`6311`] Removed Ogone-legacy payment plugin.
```

* [:backend:`6164`] Removed deprecated code (please review the instructions in the documentation
  under **Installation** > **Upgrade details to Open Forms 4.0.0** for all the necessary details):

  - Removing legacy detection and fix scripts.
  - [:backend:`4939`] Removing legacy ``zaaktype`` URL and ``documenttype`` URL fields in
    Objects API and ZGW APIs registration backends.
  - [:backend:`6294`] Using the branch number obtained through the eHerkenning login for StUF-ZDS

>     and ZGW API's, instead of the branch number retrieved from the form's submission data.

Reviewer: ```suggestion
  - [:backend:`6294`] The branch number from an eHerkenning login is now favored over the data from a
    form field if both are available.
```

**New features**

* Added data migration and import converters for fixing invalid ``date`` and ``datetime`` component
  min/max date configuration.

> * [:backend:`6351`] Refresh the list of available form components when changed the form type.

Reviewer: ```suggestion
* [:backend:`6351`] The available component types in the form designer are now updated when the form type
  is changed. 
```

* [:backend:`6281`] You can now select which document type to use for the Objects API and

>   ZGW APIs registration backends, based on the selected api group and catalogue.

Reviewer: ```suggestion
* [:backend:`6281`] You can now specify the document registration options for file upload
  components tailored to each registration backend defined on the form.
```

* [:backend:`6349`] Added import convertion, changing `is_appointment` field to `type`, for old

>   appointment forms.

Reviewer: ```suggestion
* [:backend:`6349`] Added import conversion for appointment forms to the new form type mechanism.
```

* [:backend:`6217`] Allow including the submission summary details as PDF- and/or JSON-document

>   when creating a zaak using the ZGW API's registration.

Reviewer: ```suggestion
* [:backend:`6217`] In the ZGW APIS registration, you can now add a JSON-document with the submission
  details to the case and select which file formats to relate (JSON, PDF).
```

* [:backend:`6336`] Added address registration attribute for ``addressNL`` components when using
  StUF-ZDS and ZGW APIs registration backends.
* [:backend:`5984`] Added ``contactpersoonRol`` attributes when creating rol objects through
  the ZGW APIs registration plugin.

> 

Reviewer: Additional entry:

```rst
* [:backend:`6391`] You can now easily convert simple logic rules to advanced logic.
```

> **Bugfixes**

Reviewer: Bugfix suggestions below are aligned with what's in the 3.5.4/3.4.10 patch notes where applicable.

* [:backend:`6259`] Fixed non-applicable form steps blocking form registration and breaking the

>   rendering of submission summary PDFs.

Reviewer: ```suggestion
* [:backend:`6259`] Fixed "cosign required" validation not taking into account whether a
  form step is applicable or not.
```

> * Fixed submission lookup crash in structlog.

Reviewer: ```suggestion
* Fixed missing structlog context propagation across threads, causing prefill (audit)
  logs not to be saved in the database.
```

* [:backend:`3535`] Fixed bug where pausing a form allowed users to go to the next step, even when
  progressing was blocked.

> * [:backend:`6125`] Fixed invalid empty value for ``addressNL`` components.

Reviewer: ```suggestion
* [:backend:`6125`] Fixed incorrect empty value definition for ``addressNL`` components.
```

* Fixed missing custom error message information for ``addressNL`` components.
* [:backend:`6111`] Fixed a browser window crash/freeze when editing the name of a (new)
  form step.
* [:backend:`6302`] Fixed missing logo in the confirmation PDF.
* [:backend:`6320`] Fixed missing component translations when the form does not require
  backend logic.

> 

Reviewer: Additional entries from last week:

```rst
* [:backend:`6360`] Fixed validation error for valid uploads in the new renderer when
  soft-hyphens are used, an issue similar to :backend:`5413`.
* [:backend:`6254`] Fixed pre-3.5 imports of forms that contain logic rules with
  "trigger from step" options.
* [:backend:`6389`] Fixed the form theme not being applied to cosign and form suspension
  emails.
* [:backend:`6346`] Fixed a crash in the admin when navigating to nonexistent objects that
  normally have protected upload fields.
```

**Project maintenance**

> * Reduced flakyness in CI by moving docs linkcheck to a weekly cronjob.

Reviewer: ```suggestion
* Reduced flakiness in CI by moving docs linkcheck to a weekly cronjob.
```

* Squashed form migrations until Open Forms version 3.5.
* Addressed some browser console messages.

> * Removing stale submission step validate API endpoint.

Reviewer: ```suggestion
* Removed the obsoleted submission step validate API endpoint.
```

> **Project maintenance**
> 
> * Reduced flakyness in CI by moving docs linkcheck to a weekly cronjob.
> * Squashed form migrations until Open Forms version 3.5.
> * Addressed some browser console messages.
> * Removing stale submission step validate API endpoint.
> * Updated backend dependencies:
> 
>   - django
>   - django-privates
>   - idna
>   - webob

Reviewer: ```suggestion
* Updated backend dependencies:

  - django
  - django-privates
  - idna
  - webob
  - urllib3
  - requests
  - Tornado
  - PyJWT
  - cryptography
```

+ updated frontend deps:

```rst
* Updated frontend dependencies

  - js-cookie
  - build toolchain (babel, esbuild)
  - dompurify (where possible)
```

* Updated Open Forms theming and logos.

> * Reduced flakyness in the tests.

Reviewer: ```suggestion
* Reduced flakiness in the tests.
```

* Updated documentation.

> * Using stricter component definitions in test data.

Reviewer: ```suggestion
* Test setup now uses stricter component definitions.
```

> 

Reviewer: Additional entries:

```rst
* Upgraded to Storybook 10.

```
